tests/unittest/benchmark_runner/common/oc/test_oc.py

```diff
@@ -7,58 +7,66 @@
 from benchmark_runner.common.oc.oc_exceptions import YAMLNotExist, LoginFailed, NodeNotReady
 
 
-def test_create_async_pod_yaml_not_exist():
-    """
-    This method create pod from not exist yaml
-    :return:
-    """
-    oc = OC()
-    with pytest.raises(YAMLNotExist):
-        oc.create_async(yaml=f'stressng1.yaml')
+@patch('benchmark_runner.common.oc.oc.SingletonOCLogin')  # patching login
+def test_create_async_pod_yaml_not_exist(mock_login):
+    oc = OC(kubeadmin_password="dummy-password")
+    with patch("os.path.exists", return_value=False):
+        with pytest.raises(YAMLNotExist):
+            oc.create_async(yaml='stressng1.yaml')
 
 
-def dummy_long_uuid(cls, *args, **kwargs):
-    # Your custom testing override
-    return 'bb2be20e-e4e8-57ed-a7c0-efde559b1ce2'
+# Dummy UUID for testing
+def dummy_long_uuid(self, workload):
+    return 'bb2be20e-92d5-4f10-a6ae-94f71f479d12'
 
 
-def test_short_uuid():
+@mock.patch('benchmark_runner.common.oc.oc.SingletonOCLogin')  # Prevent login
+def test_short_uuid(mock_login):
     """
-    This method test parsing long uuid
+    This method tests parsing a long UUID into a short UUID
     """
-
     with mock.patch.object(OC, 'get_long_uuid', new=dummy_long_uuid):
-        oc = OC()
-        assert oc._OC__get_short_uuid(workload='stressng_pod') == 'bb2be20e'
+        oc = OC(kubeadmin_password="dummy-password")  # Avoid LoginFailed
+        short_uuid = oc._OC__get_short_uuid(workload='stressng_pod')
+        assert short_uuid == 'bb2be20e'
+
 
+@patch('benchmark_runner.common.oc.oc.SingletonOCLogin')  # patch login
+def test_check_all_nodes_status_ready(mock_login):
+    oc = OC(kubeadmin_password="dummy-password")  # prevent LoginFailed
```

Comment:
A unit test shouldn't need a password at all, since it isn't actually logging in.

Reply:
Thats the reason I have added a patch for benchmark_runner.common.oc.oc.SingletonOCLogin